editor/store/selectors.js

```diff
@@ -95,7 +95,8 @@ export function isSavingMetaBoxes( state ) {
  * @return {boolean} Whether undo history exists.
  */
 export function hasEditorUndo( state ) {
-	return state.editor.past.length > 0;
+	const { past, present } = state.editor;
+	return past.length > 1 || last( past ) !== present;
```

Comment:
Aren't we preventing `past` from including `present`? When would those be the same?

Reply:
They should be the same at the moment an undo record is created or at the start of history. This is so that present can be overwritten. Not sure I understand the first question.